src/history/hs.c

```diff
@@ -1313,19 +1311,20 @@ __verify_history_store_id(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, uint32
         if (!found)
             WT_ERR_PANIC(session, WT_PANIC,
               "the associated history store key %s was not found in the data store %s",
-              __wt_buf_set_printable(session, hs_key->data, hs_key->size, prev_hs_key),
+              __wt_buf_set_printable(session, hs_key.data, hs_key.size, prev_hs_key),
               session->dhandle->name);
 
-        /* Swap current/previous buffers. */
-        tmp = hs_key;
-        hs_key = prev_hs_key;
-        prev_hs_key = tmp;
+        /*
+         * Copy the key memory into our scratch buffer. The key will get invalidated on our next
+         * cursor iteration.
+         */
+        WT_ERR(__wt_buf_set(session, prev_hs_key, hs_key.data, hs_key.size));
     }
     WT_ERR_NOTFOUND_OK(ret, true);
 err:
     F_CLR(&cbt->iface, WT_CURSTD_IGNORE_TOMBSTONE);
     F_CLR(hs_cursor, WT_CURSTD_IGNORE_TOMBSTONE);
-    __wt_scr_free(session, &hs_key);
+    WT_ASSERT(session, hs_key.mem == NULL && hs_key.memsize == 0);
```

Comment:
I am not sure we need this assertion here. It is now no different than other places in the code doing cursor operations. But if it is helpful I won't push harder than that.

Reply:
@sueloverso 
Hmm, my reasoning was that normally, we'd do a `__wt_buf_free` here. But I don't expect there to ever be memory to be freed. So I just asserted that I'm correct with my assumption. If I'm wrong, it would be helpful to use a scratch buffer here.

What other places in the code are you referring to?